src/views/Drawer/DrawerSidebar.js

```diff
@@ -59,26 +68,51 @@ class DrawerSidebar extends PureComponent<void, Props, void> {
     return route.routeName;
   };
 
-  _renderIcon = ({ focused, tintColor, route }: DrawerScene) => {
-    const drawer = this._getScreenConfig(route.key, 'drawer');
-    if (drawer && drawer.icon) {
-      return drawer.icon({
-        tintColor,
-        focused,
-      });
-    }
-    return null;
+  _renderIcon = (configs: DrawerConfigs) => ({ focused, route }: DrawerScene) => {
+    const drawer = focused ? configs[route.key].focused : configs[route.key].unfocused;
+    return drawer && drawer.icon ? drawer.icon : null;
   };
 
   render() {
-    const ContentComponent = this.props.contentComponent;
+    const {
+      contentComponent: ContentComponent,
+      contentOptions,
+      childNavigationProps,
+      router,
+      navigation,
+    } = this.props;
+
+    const defaultDrawerProps = ContentComponent.defaultProps || {};
+    const activeTintColor =
+      contentOptions && contentOptions.activeTintColor ?
+      contentOptions.activeTintColor :
+      defaultDrawerProps.activeTintColor;
+    const inactiveTintColor =
+      contentOptions && contentOptions.inactiveTintColor ?
+      contentOptions.inactiveTintColor :
+      defaultDrawerProps.inactiveTintColor;
+
+    const configs = navigation.state.routes.reduce((acc: DrawerConfigs, route: *) => {
```

Comment:
@ericvicenti I updated pass the router down, but I'll prefer to keep the `renderIcon` and `renderLabel` here so that it's easier to reuse if wanted without having to write the logic again.